### ISSUE TITLE
Can't install AnyEdit in STS 4.0.0.M10

### ISSUE THREAD

**Issue by ahoehma** [spring-projects/spring-ide#268]
https://marketplace.eclipse.org/content/anyedit-tools

Cannot complete the install because one or more required items could not be found.
Software currently installed: AnyEditTools 2.7.1.201709201439 (AnyEditTools.feature.group 2.7.1.201709201439)
Missing requirement: AnyEdit Tools 2.7.1.201709201439 (de.loskutov.anyedit.AnyEditTools 2.7.1.201709201439) requires 'osgi.ee; (&(osgi.ee=JavaSE)(version=1.7))' but it could not be found
Cannot satisfy dependency:
From: AnyEditTools 2.7.1.201709201439 (AnyEditTools.feature.group 2.7.1.201709201439)
To: org.eclipse.equinox.p2.iu; de.loskutov.anyedit.AnyEditTools [2.7.1.201709201439,2.7.1.201709201439]

**Comment by kdvolder** [comment]
The error says you need Java 7. What do you have?

**Comment by ahoehma** [comment]
I run my eclipse with Java 8. The author of AnyEdit give me this feedback

> nyEdit works on *all* supported Eclipse releases up to latest 4.8 nightly, and I guess that this STS thingy is simply broken: "requires 'osgi.ee; (&(osgi.ee=JavaSE)(version=1.7))' but it could not be found".

**Comment by kdvolder** [comment]
The error that you pasted means AnyEdit can't be installed because it requires Java 7. I beleave it implicitly means 'Java 7 or above' so Java 8 should be fine. Unless you are unknowingly running it with something older. This not unheard of as the Eclipse launcher works in mysterious ways and can sometimes pickup an old JRE lingering in your system somewhere.

Is that the entirety of the error message you are pasting? If not please attach the whole message so I can take a closer look.

Also, you can open the STS 'About' dialog. Go to 'installation details'. Then open 'Configuration' tab. There's a whole bunch of text there. Copy that and attach to the issue. It contains every last detail of your STS installation (more than we really need to know, but useful to check for things like, what JRE are you actually running etc.).

**Comment by martinlippert** [comment]
This is really really strange. You can install AnyEdit into Eclipse 4.7.3a and 4.8M6 without any issues, but installing it into STS4 M10 causes the mentioned error. Hmmm... Not sure what exactly is going on here. Need to investigate in more detail.

**Comment by martinlippert** [comment]
I spend quite a bit of time investigating this and it remains unclear to me what the underlying issue is at the moment. Looks like you can install it from the update site into plain Photon 4.8M6 without issues, but not into STS4 on 4.8 nor STS3 on 4.8. I think the error message about the JDK level is not correct and misleading and I've seen another one during one of my tests about missing org.eclipse.rse, even though AnyEdit doesn't seem to have any dependencies on that. I am running out of ideas how to debug this further, but will try to find a new approach to this. Will keep you updated.

**Comment by martinlippert** [comment]
It seems to be related to the way the AnyEdit update site is created, it doesn't seem to be an ordinary p2 repository, but an old-style `site.xml` based location. Not sure why, but that seems to matter. When I create a simple p2 repository for the AnyEdit feature, installation into STS works just fine.

Again, this doesn't really answer the question WHY the install fails from the old-style update location, but we are getting closer... :-)

**Comment by iloveeclipse** [comment]
There was a bug in some of 4.8 milestones related to that. If you search "anyedit" and site.xml in bugzilla you should be able to find it.

**Comment by iloveeclipse** [comment]
See https://bugs.eclipse.org/bugs/show_bug.cgi?id=530093.

**Comment by kdvolder** [comment]
Eclipse marked the bug as fixed. So I don't quite understand why it isn't working in STS builds which is are based on latest eclipse milestones where the bug is supposed to have been fixed.

**Comment by iloveeclipse** [comment]
I haven't analyzed the fix, but I believe there was some maven/p2/tycho magic required, so it can be not just Eclipse issue but also build tools issue as well.

**Comment by martinlippert** [comment]
Thanks @iloveeclipse for the pointer to the bug, that helped a lot. This is fixed now for STS4 (will be in STS4 M11) and I am going ahead and fix the same issue for STS 3.9.5 (on Photon)

**Comment by iloveeclipse** [comment]
What was the fix?

**Comment by martinlippert** [comment]
https://github.com/spring-projects/sts4/commit/2342dc8d3bfb974291883d9ed6a7a1be369b1567

**Comment by timogruetter** [comment]
I get the same error on STS 3.9.5 now... is there a workaround to get the AnyEdit Tools working? I also use Java 8. I took a look to the link above, but I don't know where I should enter the lines...

**Comment by martinlippert** [comment]
Looks like I missed to backport the fix that we put into STS4 into the STS 3.9.x stream to get it our for STS 3.9.5. Working on that now. CI build for STS 3.9.6 for early adopters will be available shortly.

**Comment by martinlippert** [comment]
Ok, the fix is in for STS 3.9.x builds, too. If you want to get a sneak peek before the next release (STS 3.9.6), you can download a CI build here: http://dist.springsource.com/snapshot/STS/nightly-distributions.html

**Comment by manueljordan** [comment]
Hello 

I just have installed two days ago through the following installer:

* `spring-tool-suite-3.9.5.RELEASE-e4.8.0-macosx-cocoa-x86_64.dmg` 

The following

```
Version: 3.9.5.RELEASE
Build Id: 201807031753
Platform: Eclipse Photon (4.8.0)
```
And I get:

![sts-anyedittools](https://user-images.githubusercontent.com/2552795/44120686-bfacfaa4-9fe2-11e8-9bd4-b8ef80876061.png)

I have the following about JRE

![sts-jre](https://user-images.githubusercontent.com/2552795/44120763-037c804c-9fe3-11e8-9fb9-60d1c958cdf0.png)

Because this issue is closed since July and I have installed it since two days ago, I assumed all should work fine.

Not sure what to do (i.e: wait the next release for STS? or wait a new version of this plugin?).

Thank you

**Comment by martinlippert** [comment]
Hey @manueljordan, as explained in the comments above (especially https://github.com/spring-projects/spring-ide/issues/268#issuecomment-404419737), the fix for this didn't made it into the STS 3.9.5 release, but will be included in STS 3.9.6 (to be released end of September). You can grab an early CI build of STS 3.9.6 that already includes the fix from the nightly download page (also mentioned above).

**Comment by manueljordan** [comment]
Sorry, I was so tired, I did not read just that comment. Thanks for the explanation.
